atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1890,8 +1890,23 @@ private void alterGcAndTombstoneInternal(
 
     public void cleanUpSchemaMutationLockTablesState() throws Exception {
         Set<TableReference> tables = lockTables.getAllLockTables();
-        dropTablesInternal(tables);
-        log.info("Dropped all schema mutation lock tables [{}]", tables.toString());
+        if (tables.size() == 0) {
+            return;
+        }
+        TableReference tableToKeep = extractFirstTable(tables);
+        if (tables.size() > 0) {
+            dropTablesInternal(tables);
+            log.info("Dropped tables [{}]", tables.toString());
+        }
+        schemaMutationLock.cleanLockState();
+        log.info("Reset the schema mutation lock in table [{}]", tableToKeep.toString());
```

Comment:
Oh. Oh... it's not a good idea to introduce such side-effects.

I would rather see something like:
```
tableToKeep = tables.first();
tablesToDrop = tables.remove(tableToKeep);
dropTablesInternal(tablesToDrop);
```